app.R

```diff
@@ -9,134 +9,31 @@ library(readr)
 library(here)
 library(purrr)
 library(dplyr)
+library(tidyverse)
 
-plot_salary_heatmap <- function(xmax, xcon){
-  
-    source <- as.data.frame(df_salary) %>%
-            filter(Age > 0 & Salary_USD <= xmax[2])
-
-    if(is.null(xcon[[1]])){
-        xcon  <- "the World"
-    } else {
-        source  <- source %>%
-            filter(Country == xcon)
-    }
-
-    x_bin_num = max(as.integer((nrow(source)/6)^(0.65)), 6)
-    y_bin_num = max(as.integer((nrow(source)/6)^(0.65) / 2), 6)
-
-    fig1 <- plot_ly(x = source$Age, y = source$Salary_USD) %>% 
-        add_histogram2d(colorscale="YlGnBu", nbinsx=x_bin_num, nbinsy=y_bin_num)
-
-
-    p <- ggplot(source,aes(x=Age)) +
-        geom_histogram(aes(y = ..density..), color="blue", fill = "blue", alpha = 0.2) +
-        geom_density(fill="blue", alpha = 0.2) +
-        theme_bw()
-        
-    fig2  <- ggplotly(p) 
-
-
-
-    fig <- subplot(
-        fig1,
-        fig2,
-        nrows = 2, 
-        margin = 0.04,
-        shareY = TRUE, 
-        titleX = TRUE
-    ) %>% layout(
-      height=350, 
-      width=450,
-      title = list(text=paste("Heatmap of ", xcon), font=list(size = 16)),
-      yaxis = list(title = 'Salary', font=list(size = 8)), 
-      legend = list(title=list(text='Counts'), font=list(size = 8))
-    )
-    
-    fig
-}
-
-plot_map <- function(xcon){
+# import salaries data
 
-    print(xcon)
+df_salary <- readr::read_csv('cleaned_salaries.csv')
```

Comment:
The csv file lives in the `data` directory. I think it would also be better to use the `here` package